### DISCUSSION TITLE
Convinent way to add image to rich text editor

### DISCUSSION THREAD

**Discussion by PegasisForever** [payloadcms/payload#574] [Feature Requests & Ideas]
Currently, you can only select already uploaded image to insert into rich text editor. To insert a new image, you need to do to the media collection, upload image, go back to rich text and insert.

Is there any way I can upload and insert an image using rich text's ui?

Also, how can I get pasting image inside rich text to work?

**Comment by jmikrut** [comment]
Hey @PegasisForever — we plan to actually release a way to be able to _**add**_ an upload right from the rich text editor, like you've suggested. This will come as soon as we get a few other features in our pipeline completed. Great question.

As far as pasting an image goes, that is a great suggestion. There may be a way for us to tackle that at the same time as the above function. Totally possible!

I've changed this discussion to a Feature Request and we'll report back soon. 👍 

If you want to take a crack at this in the meantime, I'd be happy to pair with you on it or point you in the right direction!

**Reply by ruchernchong** [reply]
Hey @jmikrut, this has been 3 years now. Any updates on this? I might have missed it if there is/are.

**Reply by shomriddho** [reply]
https://payloadcms.com/docs/rich-text/official-features#uploadfeature there is a way to add uploads right in the text block now !!!
